src/app/tasks/task-details/task-details.component.html

```diff
@@ -1,22 +1,92 @@
 <div class="col">
-    <div class="row">
-        <div class="col-3">
-            <h4 class="list-group-item-heading">{{ task?.title }} ({{task?.type?.displayName}}) </h4>
-        </div>
-        <div class="col-11">
-            <p class="list-group-item-text">{{ task?.description }}</p>
-        </div>
-        <div class="col-1">
-            <div class="col-1">
-                <button class="btn btn-danger" 
-                (click)="this.task !== null && this.task.id !== null ? onDeleted(this.task.id) : true">
-                Delete</button>
+    <form [formGroup]="editTaskForm!" (ngSubmit)="onSubmit()">
+        <div class="card card-body">
+            <div class="row">
+                <div class="col-3">
+                    <div class="row">
+                        <div class="col">
+                            <h4 class="list-group-item-heading text-nowrap">{{ task.title }} ({{task.type?.displayName}})</h4>
+                        </div>
+                        <div class="col">
+                            @if (editMode != true) {
+                                <button class="btn btn-secondary" (click)="onEdit()">Edit</button>
+                            }
+                            @else {
+                                <input class="form-control" [disabled]="!editTaskForm.valid" class="btn btn-primary" type="submit" value="Save">
+                            }
+                        </div>
+                    </div>
+                </div>
+                <div class="col-11">
+                    <p class="list-group-item-text"><span class="badge bg-primary">{{task.status?.displayName}}</span> {{ task.description }}</p>
+                </div>
+                <div class="col-1">
+                    <button class="btn btn-danger" 
+                    (click)="this.task !== null && this.task.id !== null ? onDeleted(this.task.id) : true">
+                    Delete</button>
             </div>
+            <div class="row d-flex flex-row-reverse">
+                <div class="col">
+                    <span>{{ task.createdOn.toDateString() }}</span>
+                </div>
+            </div>
+        </div>    
+        @if (this.editMode == true) {
```

Comment:
you can just leave it as @if(this.editMode) means true !this.editMode would mean false